### PR TITLE
Copy submitter to address issue 58

### PR DESCRIPTION
**Pull request by webstech**
This change is done in three commits.  It is intended to address issue #58 where the person submitting changes is not the author (this may be a simplification of the problem being addressed).

The first adds a warning comment in the /allow if a user does not have a public email address.  

The second does the same during a /submit.  It also changes the signature for gitgitgadget::submit to only pass the pull request and user info.  Passing the pull request object removes the need for the CIHelper to know what gitgitgadget needs from the pull request.

The third commit simply adds the submitter to the cc list.

### CODE REVIEW

**Comment by webstech** [comment]
Most of the testing was done using the changes in the ci-helper-test branch.  This is a basic setup to test the changes for this PR and the previous user verification PR.  Is there any interest in improving and formalizing these changes?  Just trying to get something in place for future testers.  The changes in ci-helper-test branch are  by no means complete.  Should an issue be opened to get more input?

fyi, I do not know how to use jest to mock the functions being overridden.

**Comment by webstech** [comment]
Please do not review until I make an update.  Need to authenticate with github to obtain the user email address.  Initial tests had been done after authentication was already done.

**Comment by webstech** [comment]
Okay, initial tests were not done after authentication because I do not have the key.  github-glue.ts has been updated in the first commit to authenticate.  Can't really test the code without getting a key.  The function is mocked out in testing.

Ready for review again.

File: lib/gitgitgadget.ts [hunk 1]
```diff
@@ -163,50 +164,43 @@ export class GitGitGadget {
...
         return coverMid;
     }
 
     protected async updateNotesAndPullRef(pullRequestNumber: number,
                                           additionalRef?: string):
         Promise<string> {
-        if (!await isDirectory(this.workDir)) {
-            await git(["init", "--bare", this.workDir]);
-        }
-
```
Comment (dscho) [inline lib/gitgitgadget.ts]: Is this removal intentional?

Comment (webstech) [reply lib/gitgitgadget.ts]: The init is done in the get method at line 44.  I assume that was added at some point without removing the later one.  I can restore the init if I misunderstood.

Comment (dscho) [reply lib/gitgitgadget.ts]: But that `get()` method is static, while `updateNotesAndPullRef()` is not. Therefore, the latter could be called without having had the benefit of the former guaranteeing that the `workDir` exists...

I think we need to reinstate these lines.

Comment (webstech) [reply lib/gitgitgadget.ts]: The static get() method is used to construct a new instance of GitGitGadget.  This is currently the only way to access the protected constructor so the check would have been done?

Would it be better to have the constructor do the git init and call fetchAndReReadOptions()?  This would remove the git init and fetch and readOptions() from the get() method.  It would ensure the git init has been done (and options are current) if there is a new get() alternative added to create a GitGitGadget.  This would be a separate pull request of course.

Comment (dscho) [reply lib/gitgitgadget.ts]: Ah, you're right, of course, the `get()` method is kind of the constructor: we need this to be `async`, and a constructor cannot be `async`.

So your patch is of course fine as-is, as you patiently explained to me often enough for me to understand. Thanks!
